src/libmongoc/doc/mongoc_handshake.rst

```diff
@@ -0,0 +1,89 @@
+:man_page: mongoc_add_driver_info
+
+mongoc_add_driver_info()
+=========================================
+
+Synopsis
+--------
+
+.. code-block:: c
+
+   bool
+   mongoc_add_driver_info (const char *driver_name,
```

Comment:
Apologies for back-tracking. But given @alcaeus's recent feature request of CDRIVER-4142, would it be preferable to hold off on the alias, and reduce the scope of CDRIVER-3441 to documenting `mongoc_handshake_data_append`?

Reply:
That's a good question! I actually was wondering a bit about those semantics, myself. I'm inclined to agree with your intuition as well: the motivation for /this/ PR was to provide a name that fits in better with our extant conventions, as well as document a previously undocumented function. If we'd like the "new" name to have the "new" semantics, I think that would make sense: it doesn't break old code, and the funky name should actually partially alert callers to some funky behavior. Perhaps @alcaeus' cautionary notes should be adapted into the documentation as well. 

If you both think this is a good idea, I'm for it.

Reply:
Note: The changes in this PR still seem generally useful to me-- I the change would be that I remove mongo_add_driver_info() and change the references to it back to mongoc_handshake_data_append().

Reply:
Updated as per discussion; OR we can revert the change entirely.

Reply:
I agree with removing `mongoc_add_driver_info()` in favor of only documenting `mongoc_handshake_data_append()`. I have updated the description of CDRIVER-3441.